### PR TITLE
Replace all whitespace in Text nodes with a single space (fixes #164)

### LINKED ISSUE #164: Vue Native doesn't strip (leading) whitespace in multi-line Text elements
Steps to reproduce:

- `vue-native init VueNativeText --no-crna`
- `cd VueNativeText`
- Edit App.vue to add a text element with the contents split over multiple lines, as seen below.
- `react-native run-ios`

Here are the contents of my App.vue :
```
<template>
  <view class="container">
    <text class="text-color-primary">My Vue Native App</text>
    <text class="wide">some text
                                on multiple
                                           lines
    </text>
  </view>
</template>

<style>
.container {
  background-color: white;
  align-items: center;
  justify-content: center;
  flex: 1;
}

.text-color-primary {
  color: blue;
}

.wide {
  border-width: 1px;
  width: 90%;
}
</style>
```

Here is a screenshot of what appears in the app:
Screen Shot 2019-07-08 at 11 32 21 AM

This doesn't happen in a React Native app; multiple spaces are stripped out and multiple lines are collapsed and wrapped.

You could argue that being able to control when a new line is started is useful (given that I have seen a number of answers for how to *add* newlines to React Native Text elements), however compatibility with React Native is probably better.  At the very least, the leading whitespace should be removed.

### PR DESCRIPTION
**Pull request by RohanTalip**
According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions \s includes \u2028 and \u2029 .

Fixes issue #164.

### CODE REVIEW

**Comment by RohanTalip** [comment]
FYI, I rebased from the master branch and added the source changes.

**Comment by RishabhKarnad** [comment]
Hi @RohanTalip. As a result of this change, even whitespace characters like `&nbsp;` will be ignored, which is probably unintended (`&nbsp;` has effects in both React Native and Vue.js).

According to the Vue.js issues [3895](https://github.com/vuejs/vue/issues/3895) and [4268](https://github.com/vuejs/vue/issues/4268), the 2028 and 2029 unicode characters are treated specially because otherwise, the ES compiler treats them as actual line separators in the code, leading to a runtime error.

So basically, I think the `transformSpecialNewlines` function should be left as it is.

I think we'll need to take a closer look at how Vue.js handles whitespace (it is almost certainly done in the HTML parser) and why it doesn't work in Vue Native.

**Comment by RohanTalip** [comment]
> As a result of this change, even whitespace characters like `&nbsp;` will be ignored, which is probably unintended (`&nbsp;` has effects in both React Native and Vue.js).

Good point.  (\s includes \u00a0, which is the non-breaking space in Unicode.)


> I think we'll need to take a closer look at how Vue.js handles whitespace (it is almost certainly done in the HTML parser) and why it doesn't work in Vue Native.

Good luck!

In the meantime, @RishabhKarnad, does it make sense for me to create another pull request that keeps `transformSpecialNewlines` and simply replaces spaces, tabs and newline characters with a single space?  i.e. :
```
    var text = transformSpecialNewlines(ast.text);
    text = text.replace(/[ \t\n]+/g, " ");
    return JSON.stringify(text);
```

Maybe also include form feed characters? (`\f`)

**Comment by RishabhKarnad** [comment]
@RohanTalip sounds good (`\f` can also be included; I tried using it inside Vue templates and Vue seems to ignore them)
